cdap-api/src/main/java/co/cask/cdap/api/schedule/ProgramStatusTriggerInfo.java

```diff
@@ -0,0 +1,128 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.schedule;
+
+import co.cask.cdap.api.ProgramStatus;
+import co.cask.cdap.api.app.ApplicationSpecification;
+import co.cask.cdap.api.app.ProgramType;
+import co.cask.cdap.api.workflow.WorkflowToken;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import javax.annotation.Nullable;
+
+/**
+ * The program status trigger information to be passed to the triggered program.
+ */
+public class ProgramStatusTriggerInfo extends TriggerInfo {
+
+  private final String namespace;
+  private final ApplicationSpecification applicationSpecification;
+  private final ProgramType programType;
+  private final String program;
+  private final Set<ProgramStatus> triggerStatues;
+  private final String runId;
+  private final ProgramStatus programStatus;
+  @Nullable
+  private final WorkflowToken workflowToken;
+  private final Map<String, String> runtimeArguments;
+
+  public ProgramStatusTriggerInfo(String namespace, ApplicationSpecification applicationSpecification,
+                                  ProgramType programType, String program, Set<ProgramStatus> triggerStatues,
+                                  String runId, ProgramStatus programStatus,
+                                  @Nullable WorkflowToken workflowToken,
+                                  Map<String, String> runtimeArguments) {
+    super(Trigger.Type.PROGRAM_STATUS);
+    this.namespace = namespace;
+    this.applicationSpecification = applicationSpecification;
+    this.programType = programType;
+    this.program = program;
+    this.triggerStatues = Collections.unmodifiableSet(new HashSet<>(triggerStatues));
+    this.runId = runId;
+    this.programStatus = programStatus;
+    this.workflowToken = workflowToken;
+    this.runtimeArguments = Collections.unmodifiableMap(new HashMap<>(runtimeArguments));
+  }
+
+  /**
+   * @return The namespace of the triggering program.
+   */
+  public String getNamespace() {
+    return namespace;
+  }
+
+  /**
+   * @return The application specification of the application that contains the triggering program.
+   */
+  public ApplicationSpecification getApplicationSpecification() {
+    return applicationSpecification;
+  }
+
+  /**
+   * @return The program type of the triggering program.
+   */
+  public ProgramType getProgramType() {
+    return programType;
+  }
+
+  /**
+   * @return The program name of the triggering program.
+   */
+  public String getProgram() {
+    return program;
+  }
+
+  /**
+   * @return An immutable set of all the program statuses that can satisfy the program status trigger.
+   */
+  public Set<ProgramStatus> getTriggerStatues() {
```

Comment:
What is this for?? The triggering info should only contains information of what actually happened. I think we only need the `getProgramStatus` below.